labs/b9.md

```diff
@@ -0,0 +1,178 @@
+
+What is this doing? We first specify a base image `fedora:latest` (Fedora 27). We then specify that we should run (`RUN`) the command `yum update` (`yum` is Fedora's package manager), and then `yum install pip3` installs `pip` for `python3`. Then we install the `pyfiglet` and `termcolor` packages via `pip3` since our python script depends on them. We add the script for our machine to the container, and lastly we run the script with `python3.6`.   Take a moment to appreciate how cool this is. We have a completely different Linux distribution running on our system that ships with a version of python that is not in `apt` on your student VMs!
+
+Build the `Dockerfile` with the following:
+
+`docker build -t missile:latest .`    
+
+This tells Docker to look in the current directory for a `Dockerfile` to build, and builds it. The `-t` flag tells Docker to tag this build with the name `missile:latest`. Note that building the missile image will take a couple of minutes to complete.      
+
+You can see all of the images you’ve built on your machine with the `docker images` command.    
+
+##### Questions    
+
+1. Run the image you just built. As before, don’t forget the `-it` flags, and you can exit using CTRL+D. What do you observe?    
+2. Write and build a `Dockerfile` based on `ubuntu:xenial` that installs the packages `fortune` and `fortunes-min` and runs the fortune executable (located in `/usr/games/fortune` after you install it). Note that you won’t need to use the `-it` flags when you run the container as fortune doesn’t need `STDIN`. Submit your `Dockerfile` with this lab.  Hint: if you’re having trouble writing your Dockerfile, try booting an interactive container and installing both packages. How can you translate what you did interactively to a Dockerfile?   
+3. Paste the output running `docker images` command after questions 1 and 2.      
+
+#### Dockerizing a Web Application    
+
+For our last trick, we’re going to use Docker to run multiple Apache web servers inside containers. 
+
+For simplicity, you will not have to write this `Dockerfile`. Go ahead and pull the `httpd` image from Docker Hub. Now, it’s your job to figure out how to run three instances of the Apache containers on your machine.           
+
+Docker creates a separate network for containers, so you will need to forward your host port to your container’s port (this is called [port forwarding](https://en.wikipedia.org/wiki/Port_forwarding), or port mapping). The container is listening on port 80 by default. It is your job to run each instance on ports 5000, 5001, and 5002.  I recommend running the containers in detached mode with the `-d` flag.
+
+Hints (if you want them): 
+- Don’t expect to see any output until you visit the web application from your host machine.
+- The `-p` flag takes in a colon separated pair of HOST\_PORT:CONTAINER\_PORT (it can actually accept a ton of more options, but don’t worry about that for now). 
```

Comment:
I think putting `HOST_PORT:CONTAINER_PORT` in backticks would be good, then you don't have to escape the underscores too.